### PR TITLE
docs(samples): Various ways of creating a new disk

### PR DESCRIPTION
**Pull request by m-strzelczyk**
Will go to this [doc site](https://cloud.google.com/compute/docs/disks/create-disk-from-source)

### CODE REVIEW

File: samples/snippets/disks/create_kms_encrypted_disk.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            For example: "zones/us-west3-b/diskTypes/pd-ssd"
+        disk_size_gb: size of the new disk in gigabytes
+        kms_key_name: URL of the key from KMS. The key might be from another project, as
+            long as you have access to it. The data will be encrypted with the same key
+            in the new disk. This value uses following format:
+            "projects/{kms_project_id}/locations/{region}/keyRings/{key_ring}/cryptoKeys/{key}"
+        disk_link: a link to the disk you want to use as a source for the new disk.
+            This value uses the following format: "projects/{project_name}/zones/{zone}/disks/{disk_name}"
+        image_link: a link to the image you want to use as a source for the new disk.
+            This value uses the following format: "projects/{project_name}/zones/{zone}/disks/{disk_name}"
```
Comment (parthea) [inline samples/snippets/disks/create_kms_encrypted_disk.py:106]: Can you confirm that the example for `image_link` is correct? Based on [these docs](https://cloud.google.com/compute/docs/reference/rest/v1/disks/insert), should it be something like `global/images/family/my-image-family`?

File: samples/snippets/disks/regional_delete.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    return result
+
+
+def delete_regional_disk(project_id: str, region: str, disk_name: str) -> NoReturn:
+    """
+    Deletes a disk from a project.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the region in which is the disk you want to delete.
```
Comment (parthea) [inline samples/snippets/disks/regional_delete.py:83]:
```suggestion
        region: name of the region in which is the disk you want to delete.
```

**Review by parthea — APPROVED** [review]
Added minor observations, otherwise LGTM

File: samples/snippets/disks/create_kms_encrypted_disk.py (hunk 1 above)
Comment (m-strzelczyk) [reply samples/snippets/disks/create_kms_encrypted_disk.py:106]: You're right, will fix it.

File: samples/snippets/disks/regional_delete.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+    if operation.warnings:
+        print(f"Warnings during {verbose_name}:\n", file=sys.stderr, flush=True)
+        for warning in operation.warnings:
+            print(f" - {warning.code}: {warning.message}", file=sys.stderr, flush=True)
+
+    return result
+
+
+def delete_regional_disk(project_id: str, region: str, disk_name: str) -> NoReturn:
```
Comment (parthea) [inline samples/snippets/disks/regional_delete.py:77]:
```suggestion
def delete_regional_disk(project_id: str, zone: str, disk_name: str) -> NoReturn:
```
`region` should be changed to `zone` to match the docstring below.
```
 zone: name of the region in which is the disk you want to delete.
```

Comment (m-strzelczyk) [reply samples/snippets/disks/regional_delete.py:77]: It's the docstring that is wrong in this case :) It's a region we want here.

File: samples/ingredients/disks/create_kms_encrypted_disk.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+from google.cloud import compute_v1
+
+
+# <INGREDIENT create_kms_encrypted_disk>
+def create_kms_encrypted_disk(project_id: str, zone: str, disk_name: str, disk_type: str,
+                              disk_size_gb: int, kms_key_name: str,
+                              disk_link: Optional[str] = None, image_link: Optional[str] = None) -> compute_v1.Disk:
+    """
+    Creates a new disk in a project in given zone. If disk_link and image_link are not provided, the disk will be
```
Comment (savijatv) [inline samples/ingredients/disks/create_kms_encrypted_disk.py:29]: Creates a zonal disk in a project. If you do not provide values for `disk_link` or `image_link`, an empty disk will be created.

Maciej, isn't it either source disk or source image? Hence suggested 'or' in the above comment.

File: samples/ingredients/disks/create_kms_encrypted_disk.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            long as you have access to it. The data will be encrypted with the same key
+            in the new disk. This value uses following format:
+            "projects/{kms_project_id}/locations/{region}/keyRings/{key_ring}/cryptoKeys/{key}"
+        disk_link: a link to the disk you want to use as a source for the new disk.
+            This value uses the following format: "projects/{project_name}/zones/{zone}/disks/{disk_name}"
+        image_link: a link to the image you want to use as a source for the new disk.
+            This value uses the following format: "projects/{project_name}/global/images/{image_name}"
+
+    Returns:
+        An unattached Disk instance.
```
Comment (savijatv) [inline samples/ingredients/disks/create_kms_encrypted_disk.py:50]: An attachable disk.

File: samples/ingredients/disks/regional_create_from_source.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+
+# <INGREDIENT create_regional_disk_from_disk>
+def create_regional_disk(project_id: str, region: str, replica_zones: Iterable[str],
+                         disk_name: str, disk_type: str,
+                         disk_size_gb: int,
+                         disk_link: Optional[str] = None,
+                         snapshot_link: Optional[str] = None) -> compute_v1.Disk:
+    """
+    Creates a new regional disk in a project in given zone with a zonal disk as a
```
Comment (savijatv) [inline samples/ingredients/disks/regional_create_from_source.py:31]: Creates a regional disk from an existing zonal disk in a given project.

File: samples/ingredients/disks/regional_create_from_source.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+            "regions/{region}/diskTypes/(pd-standard|pd-ssd|pd-balanced|pd-extreme)".
+            For example: "regions/us-west3/diskTypes/pd-ssd"
+        disk_size_gb: size of the new disk in gigabytes
+        disk_link: a link to the disk you want to use as a source for the new disk.
+            This value uses the following format: "projects/{project_name}/zones/{zone}/disks/{disk_name}"
+        snapshot_link: a link to the snapshot you want to use as a source for the new disk.
+            This value uses the following format: "projects/{project_name}/global/snapshots/{snapshot_name}"
+
+    Returns:
+        An unattached Disk instance.
```
Comment (savijatv) [inline samples/ingredients/disks/regional_create_from_source.py:51]: An attachable regional disk ?

File: samples/ingredients/disks/regional_delete.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+# <INGREDIENT delete_regional_disk>
+def delete_regional_disk(project_id: str, region: str, disk_name: str) -> NoReturn:
+    """
+    Deletes a disk from a project.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        region: name of the region in which is the disk you want to delete.
+        disk_name: name of the disk you want to delete.
```
Comment (savijatv) [inline samples/ingredients/disks/regional_delete.py:33]: name of the disk that you want to delete.

File: samples/ingredients/disks/regional_delete.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+
+# <INGREDIENT delete_regional_disk>
+def delete_regional_disk(project_id: str, region: str, disk_name: str) -> NoReturn:
+    """
+    Deletes a disk from a project.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        region: name of the region in which is the disk you want to delete.
```
Comment (savijatv) [inline samples/ingredients/disks/regional_delete.py:32]: name of the region where the disk is located.

File: samples/ingredients/disks/create_kms_encrypted_disk.py (hunk 4 above)
Comment (m-strzelczyk) [reply samples/ingredients/disks/create_kms_encrypted_disk.py:29]: Yes, those two are mutually exclusive. `or` fits better here.

**Pull request merged**